### ISSUE TITLE
Feature: Date format follows system settings

### ISSUE THREAD

**Issue by DeepChirp** [AutoDarkMode/Windows-Auto-Night-Mode#786]
**Describe the enhancement or feature you'd like**
I think date format should follow system settings. I am in Chinese Mainland, and the date format here is YYYY-MM-DD. However, the current date format in the software is DD-MM-YYYY, which makes me very uncomfortable. So I hope the software can detect the date format of the system and apply it to the software.
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/66902050/276969a1-ae4d-4090-a1fd-201cea76365a)

**Describe alternatives you've considered**
_None_

**Additional context**
_None_

**Comment by Armin2208** [comment]
You are right, this should and will be fixed in future

**Comment by Spiritreader** [comment]
This is the location in codee for improving this behavior.

https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/blob/aa6bf4ea80109204a5222ad1980339008358d935/AutoDarkModeApp/Pages/PageTime.xaml.cs#L611

Instead of blanket-setting the locale to the "de" culture, we'd have to replace this by retrieving the currentculture.

Should be a pretty easy fix for anyone who's interested in tackling it until I get around to fixing and testing it (which can be quite some time because I have very little time to work on ADM nowadays and most of it goes into ensuring things don't completely break).

**Comment by medic17** [comment]
> This is the location in codee for improving this behavior.
> 
> https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/blob/aa6bf4ea80109204a5222ad1980339008358d935/AutoDarkModeApp/Pages/PageTime.xaml.cs#L611
> 
> Instead of blanket-setting the locale to the "de" culture, we'd have to replace this by retrieving the currentculture.
> 
> Should be a pretty easy fix for anyone who's interested in tackling it until I get around to fixing and testing it (which can be quite some time because I have very little time to work on ADM nowadays and most of it goes into ensuring things don't completely break).

It's not as simple as that. After some testing it seems that `CurrentCulture` does not use the actual regional settings from the users system. I debugged for quite a bit and found that it is being overridden by LanguageHelper() which for some reason is defined twice

**Comment by Spiritreader** [comment]
It's defined twice because we're I using different UI frameworks for Win10 and Win11.

In theory it should be possible to set a culture for specific fields

**Comment by medic17** [comment]
> It's defined twice because we're I using different UI frameworks for Win10 and Win11.
> 
> In theory it should be possible to set a culture for specific fields

Why do you need to touch `CultureInfo` at all? Couldn't we just set just `CultureUIinfo`?

**Comment by Spiritreader** [comment]
> > It's defined twice because we're I using different UI frameworks for Win10 and Win11.
> > In theory it should be possible to set a culture for specific fields
> 
> Why do you need to touch `CultureInfo` at all? Couldn't we just set just `CurrentUICulture`?

Maybe? Needs to be tested. 
I faintly remember having issues with the culture correctly applying.
The UI is something I kind of inherited maintaining so a lot of the specifics are something I haven't investigated closely when adding new stuff

**Comment by Jay-o-Way** [comment]
https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.datetimeformat?view=net-9.0

**Comment by ChenYiLins** [comment]
> Instead of blanket-setting the locale to the "de" culture, we'd have to replace this by retrieving the currentculture.

So should we make the 12-hour clock follow the system settings, or keep it as it is now, and let users choose whether to use the 12-hour clock in ADM. What's everyone's opinion?

**Comment by Jay-o-Way** [comment]
My personal opinion is: "Why does this setting even exist?". An app should "simply" follow system/local settings 🙂

**Comment by ChenYiLins** [comment]
> My personal opinion is: "Why does this setting even exist?". An app should "simply" follow system/local settings 🙂 Reminds me: this should also apply to **TimePicker** controls 💡

I agree with you very much!!! I think it is more important to do subtraction than to add points in many cases. For example, in this scenario, adding a 12-hour option to the configuration file is an inconspicuous point, but inadvertently, many codes become coupled, which is a headache in later modification and maintenance. (For example, if we add a time-related function, we have to consider whether to use a 12-hour clock or a 24-hour clock for it.) I initiated a poll on TG channel, and more users prefer the system choice.

**Comment by Armin2208** [comment]
> My personal opinion is: "Why does this setting even exist?". An app should "simply" follow system/local settings 🙂 Reminds me: this should also apply to **TimePicker** controls 💡

ADM currently follows the local settings but still gives the user the option to change this behavior. I think that's a good practice.

**Comment by Jay-o-Way** [comment]
There's more to localization than only the dates and times. Also the ~format~ decimal symbol of the latitude longitude: `.` or `,`. Because the _website_ uses a period, but the **NumberBox** expects the local decimal delimiter, which is a `,` for me. When I paste the value `5.056375` and Enter, the NumberBox resets to zero.

**Comment by ChenYiLins** [comment]
> There's more to localization than only the dates and times. Also the ~format~ decimal symbol of the latitude longitude: `.` 或 `,`. Because the _website_ uses a period, but the **NumberBox** expects the local decimal delimiter, which is a `,` for me. When I paste the value `5.056375` and Enter, the NumberBox resets to zero.

Things are complicated now. At first, I thought NumberFormatter could solve the problem simply. winui's NumberBox has undergone many improvements, and I found a lot of issues in github. The previous NumberBox didn't even support the use of `,` as separators. The good news is that it is supported now, and the bad news is that when it is supported,`.` cannot be parsed. I'm really mad. I tried to implement INumberParser, but I found that we couldn't replace it at all. At present, the scheme I think of is to use TextBox instead of NumberBox ╰（‵□′）╯

**Comment by Jay-o-Way** [comment]
TimePickers are not perfect yet. There's a unwanted delay in the UI format because of `await LoadGeolocationData();`

![Image](https://github.com/user-attachments/assets/f9e4765c-bc31-4c78-a550-9d19d0ec4e2f)

While the strings are great now 😄

![Image](https://github.com/user-attachments/assets/83ebe162-ae25-49e5-9693-058171450567)

**Comment by ChenYiLins** [comment]
> TimePickers are not perfect yet. There's a unwanted delay in the UI format because of `await LoadGeolocationData();` and possibly `Task.Delay(1000)`. We should de-couple the "12HourFormat" check.

Yep! U are right. I tried to load TimePick's properties first, and now it seems that there should be no problem.

> Also, we should cache location data. Because retrieving this every time may be very slow to respond. Maybe even add a "Refresh" button and?

I think it's a good idea for this. Because the retrieval speed seems to depend on Microsoft's server. Many times I don't feel the delay here, but this is definitely a more user-friendly way to deal with it.

**Comment by Jay-o-Way** [comment]
> Because the retrieval speed seems to depend on Microsoft's server

And, of course, slow WiFi 😅

**Comment by Jay-o-Way** [comment]
@DeepChirp we have updated a number of UI things so that it will be more compliant. The release of the new version will definitely take some more time, but I'll close this issue for now. Thanks for the report!
